### PR TITLE
Add short section on LLVM sanitizer usage

### PR DESCRIPTION
**Pull request by hassila**
Added llvm-sanitizers.md with a reference from the top page.

### CODE REVIEW

File: llvm-sanitizers.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# LLVM TSAN / ASAN
+
+For multithreaded and low-level unsafe interfacing server code, the ability to use LLVM:s [ThreadSanitizer](https://clang.llvm.org/docs/ThreadSanitizer.html) and  
+[AddressSanitizer](https://clang.llvm.org/docs/AddressSanitizer.html) can help troubleshoot invalid thread usage and invalid usage/access of memory.
+
+There is a [blog post](https://swift.org/blog/tsan-support-on-linux/) outlining the usage of TSAN.
+
+The short story is to use the swiftc command line options `-sanitize=address` and `-santize=thread` to each respective tool.
```
Comment (weissi) [inline llvm-sanitizers.md:8]: we should probably advertise SwiftPM's `--sanitize=address/thread`. Ie `swift build --sanitize=thread` for example?

Comment (hassila) [reply llvm-sanitizers.md:8]: Sure, will add.

**Comment by hassila** [comment]
Updated.

File: llvm-sanitizers.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Also for Swift Package Manager projects you can use `--sanitize` at the command line, e.g.:
+
+`swift build --sanitize=address myProgram`
```
Comment (weissi) [inline llvm-sanitizers.md:12]:
```suggestion
    swift build --sanitize=address
```

File: llvm-sanitizers.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+or
+
+`swift build --sanitize=thread myProgram`
```
Comment (weissi) [inline llvm-sanitizers.md:16]:
```suggestion
    swift build --sanitize=thread
```

File: llvm-sanitizers.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+and it can be used for the tests too:
+
+`swift test --sanitize=address myProgram`
```
Comment (weissi) [inline llvm-sanitizers.md:20]:
```suggestion
    swift test --sanitize=address
```

File: llvm-sanitizers.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+or
+
+`swift test --sanitize=thread myProgram`
```
Comment (weissi) [inline llvm-sanitizers.md:24]:
```suggestion
    swift test --sanitize=thread
```

**Review by weissi — COMMENTED** [review]
Swift test / build usually don't take "myProgram" as arguments, let's remove them?

File: llvm-sanitizers.md (hunk 2 above)
Comment (weissi) [reply llvm-sanitizers.md:12]: also the four spaces should give you a nice code block here

**Comment by hassila** [comment]
> Swift test / build usually don't take "myProgram" as arguments, let's remove them?

Done!

**Review by weissi — APPROVED** [review]
awesome, thanks!

File: llvm-sanitizers.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+For multithreaded and low-level unsafe interfacing server code, the ability to use LLVM:s [ThreadSanitizer](https://clang.llvm.org/docs/ThreadSanitizer.html) and  
+[AddressSanitizer](https://clang.llvm.org/docs/AddressSanitizer.html) can help troubleshoot invalid thread usage and invalid usage/access of memory.
+
+There is a [blog post](https://swift.org/blog/tsan-support-on-linux/) outlining the usage of TSAN.
+
+The short story is to use the swiftc command line options `-sanitize=address` and `-santize=thread` to each respective tool.
+
+Also for Swift Package Manager projects you can use `--sanitize` at the command line, e.g.:
+
+    swift build --sanitize=address
```
Comment (tomerd) [inline llvm-sanitizers.md:12]: make into code snippet?

```bash
swift build --sanitize=address
```

File: llvm-sanitizers.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+    swift build --sanitize=thread
+
+and it can be used for the tests too:
+
+    swift test --sanitize=address
+
+or
+
+    swift test --sanitize=thread
```
Comment (tomerd) [inline llvm-sanitizers.md:24]: ditto ^^

**Review by tomerd — APPROVED** [review]
this is great, one idea for improvement

**Review by ktoso — APPROVED** [review]
Very nice! Thank you; fixed formatting

File: llvm-sanitizers.md (hunk 6 above)
Comment (weissi) [reply llvm-sanitizers.md]: Four spaces make code too. See here:

    This is just indented with for spaces

whereas

```
that is done using treble backticks
```

.

Comment (ktoso) [reply llvm-sanitizers.md]: Yeah, it works though I'm not sure about adopting it throughout... it's a bit easier to format code consistently with the ``` fences (and review if indented correctly etc)

**Pull request merged**